argus/client/base.py

```diff
@@ -0,0 +1,170 @@
+import re
+from dataclasses import asdict
+from typing import Any, Type
+from uuid import UUID
+
+import requests
+
+from argus.backend.util.enums import TestStatus
+from argus.client.sct.types import LogLink
+
+JSON = dict[str, Any] | list[Any] | int | str | float | bool | Type[None]
+
+
+class ArgusClientError(Exception):
+    pass
+
+
+class ArgusClient:
+    schema_version: str | None = None
+
+    class Routes():
+        # pylint: disable=too-few-public-methods
+        SUBMIT = "/testrun/$type/submit"
+        HEARTBEAT = "/testrun/$type/$id/heartbeat"
+        SET_STATUS = "/testrun/$type/$id/set_status"
+        SET_PRODUCT_VERSION = "/testrun/$type/$id/update_product_version"
+        SUBMIT_LOGS = "/testrun/$type/$id/logs/submit"
+        FINALIZE = "/testrun/$type/$id/finalize"
+
+    def __init__(self, auth_token: str, base_url: str, api_version="v1") -> None:
+        self._auth_token = auth_token
+        self._base_url = base_url
+        self._api_ver = api_version
+
+    @property
+    def auth_token(self) -> str:
+        return self._auth_token
+
+    @staticmethod
+    def verify_location_params(endpoint: str, location_params: dict[str, str]) -> bool:
+        required_params: list[str] = re.findall(r"\$[\w_]+", endpoint)
+        for param in required_params:
+            if param.lstrip("$") not in location_params.keys():
+                raise ArgusClientError(f"Missing required location argument for endpoint {endpoint}: {param}")
+
+        return True
+
+    @staticmethod
+    def check_response(response: requests.Response, expected_code: int = 200) -> JSON:
+        if response.status_code != expected_code:
+            raise ArgusClientError(
+                f"Unexpected HTTP Response encountered - expected: {expected_code}, got: {response.status_code}",
+                expected_code,
```

Comment:
why provide expected_code and status_code args to exception? It is shown in message.

Reply:
This is so that in case you want to propagate the exception somewhere else by raising another exception you have access to the affected fields without parsing the message.